File: go/internal/scoring/testdata/structural-v1/typescript/state_coherent_after.ts
export class CoherentAfter {
  private total = 0;

  private advanceState(delta: number): number {
    if (delta < 0) return this.total;
    this.total += delta;
    if (this.total > 100) this.total = 100;
    return this.total;
  }

  advance(delta: number): number { return this.advanceState(delta); }
  advanceAgain(delta: number): number { return this.advanceState(delta); }
}

File: go/internal/scoring/testdata/structural-v1/rust/state_coherent_after.rs
pub struct CoherentAfter { total: i32 }

impl CoherentAfter {
    fn advance_state(&mut self, delta: i32) -> i32 {
        if delta < 0 { return self.total; }
        self.total += delta;
        if self.total > 100 { self.total = 100; }
        self.total
    }

    pub fn advance(&mut self, delta: i32) -> i32 { self.advance_state(delta) }
    pub fn advance_again(&mut self, delta: i32) -> i32 { self.advance_state(delta) }
}
